File: DataStructure/programmers_062.py
def solution(s):
    answer = 0
    for i in range(len(s)):
        s = s[1:] + s[:1]
        st = []
        for i in s:
            if i in '({[':
                st.append(i)
            elif st:
                if st[-1] == '(' and i == ')':
                    st.pop()
                elif st[-1] == '[' and i == ']':
                    st.pop()
                elif st[-1] == '{' and i == '}':
                    st.pop()
                else:
                    break
            else:
                st.append(i)
                break
                
        if len(st) == 0:
            answer += 1
    return answer

File: JavaScript/programmers_062.js
// LV.2 짝지어 제거하기

function solution(s) {
  var answer = -1;
  var arr = s.split("");
  var stack = [arr[0]];
  for (let i = 1; i < arr.length; i++) {
    if (stack.length === 0) {
      stack.push(arr[i]);
    } else {
      var comp = stack.pop();
      if (comp !== arr[i]) {
        stack.push(comp);
        stack.push(arr[i]);
      }
    }
  }
  if (stack.length === 0) answer = 1;
  else answer = 0;
  return answer;
}
